### PR TITLE
[19.05] Listen to history changes in tool form until job has been submitted

### LINKED ISSUE #7940: uploading a new file does not refresh the tool form
Hi,

The tool page is no longer refreshed (release 19-01) after a new file has been added in history.
ie:
from an empty history, if you choose Datamash (operations on tabular data).
> no tabular dataset available

then upload a tabular file

>  no tabular dataset available

With galaxy 17-09, the tool page is refreshed, and Datamash is ready to go with the newly uploaded dataset.

### PR DESCRIPTION
**Pull request by guerler**
Fixes #7940.

### CODE REVIEW

File: client/galaxy/scripts/mvc/tool/tool-form-base.js [hunk 1]
```diff
@@ -79,11 +79,6 @@ export default FormBase.extend({
                 self.deferred.reset();
                 self.deferred.execute(process => {
                     self.model.get("postchange")(process, self);
-                    if (self.model.get("listen_to_history")) {
```
Comment (mvdbeek) [inline client/galaxy/scripts/mvc/tool/tool-form-base.js:82]: Galaxy is now unused, and the linter doesn't like that (https://circleci.com/gh/galaxyproject/galaxy/15416?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link)

**Review by mvdbeek — APPROVED** [review]
Works great, that's a nice usability fix. Might want to also target 19.01 ?

**Comment by guerler** [comment]
@galaxybot test this

**Comment by mvdbeek** [comment]
Don't worry about the failing tests, Jenkins has some hiccups when switching branches. If circle-ci and selenium pass it's OK for me.

**Comment by martenson** [comment]
It takes one person to backport now, so we need not to be worried about retargetting too much I think.

**Comment by FredericBGA** [comment]
Thank you @guerler

**Pull request merged**